jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java

```diff
@@ -59,6 +60,18 @@ private static boolean isHttpsProtocol(URL url) {
     return "https".equals(url.getProtocol());
   }
 
+  // https://github.com/GoogleContainerTools/jib/issues/1316
+  @VisibleForTesting
+  static boolean isBrokenPipe(IOException original) {
+    for (Throwable exception = original; exception != null; exception = exception.getCause()) {
```

Comment:
Would it be possible for an exception's cause to be the original exception and cause an infinite loop here?

Reply:
`getCause()` returns null if the cause is itself, but its Javadoc doesn't say that explicitly. I think it's good to guard against an infinite loop.

```java
    public synchronized Throwable getCause() {
        return (cause==this ? null : cause);
    }
```